swarm/src/connection.rs

```diff
@@ -344,18 +352,18 @@ where
                 }
             }
 
-            // Compute new shutdown
-            if let Some(new_shutdown) =
+            if let Some(new_timeout) =
                 compute_new_shutdown(handler.connection_keep_alive(), shutdown, *idle_timeout)
```

Comment:
I think you were right about having this one _inside_ the `if` below. We shouldn't ask the handler for its keep-alive unless the connection is actually idle.

Reply:
fixed `77b840e`